DSLink.Tests/ResponderTests.cs

```diff
@@ -0,0 +1,231 @@
+            var response = responses[0];
+            var updates = response["updates"];
+
+            Assert.AreEqual(1, response["rid"].Value<int>());
+            Assert.AreEqual("open", response["stream"].Value<string>());
+
+            Assert.IsTrue(JToken.DeepEquals(updates[0], new JArray
+            {
+                "$is",
+                "node"
+            }));
+
+            var testValueUpdate = updates[1][1];
+            Assert.AreEqual("testValue", updates[1][0].Value<string>());
+            Assert.NotNull(testValueUpdate["$is"]);
+            Assert.NotNull(testValueUpdate["$type"]);
+            Assert.NotNull(testValueUpdate["value"]);
+            Assert.NotNull(testValueUpdate["ts"]);
+            Assert.AreEqual("node", testValueUpdate["$is"].Value<string>());
+            Assert.AreEqual("number", testValueUpdate["$type"].Value<string>());
+            Assert.AreEqual(123, testValueUpdate["value"].Value<int>());
+            Assert.AreEqual(JTokenType.String, testValueUpdate["ts"].Type);
+
+            var testNodeUpdate = updates[2][1];
+            Assert.AreEqual("testNodeConfigs", updates[2][0].Value<string>());
+            Assert.NotNull(testNodeUpdate["$is"]);
+            Assert.NotNull(testNodeUpdate["$testString"]);
+            Assert.NotNull(testNodeUpdate["$testNumber"]);
+            Assert.AreEqual("node", testNodeUpdate["$is"].Value<string>());
+            Assert.AreEqual("string", testNodeUpdate["$testString"].Value<string>());
+            Assert.AreEqual(123, testNodeUpdate["$testNumber"].Value<int>());
+
+            Console.WriteLine(updates.ToString());
+        }
+
+        [Test]
+        public async Task TestInvoke()
+        {
+            bool actionInvoked = false;
+            _responder.SuperRoot.CreateChild("testAction")
+                .SetInvokable(Permission.Write)
+                .SetAction(new ActionHandler(Permission.Write, async (request) =>
+                {
+                    Assert.AreEqual("string", request.Parameters["testString"].Value<string>());
+                    Assert.AreEqual(123, request.Parameters["testNumber"].Value<int>());
+                    actionInvoked = true;
+                    await request.Close();
+                }))
+                .BuildNode();
+
+            await _invokeNode();
+
+            Assert.IsTrue(actionInvoked);
+        }
+
+        [Test]
+        public async Task TestSubscribe()
+        {
+            var requestResponses = await _subscribeToNode();
+
+            var requestUpdate = requestResponses[0];
+            var update = requestUpdate["updates"][0];
+            var requestClose = requestResponses[1];
+
+            // Test for subscribe method value update.
+            Assert.AreEqual(0, requestUpdate["rid"].Value<int>()); // Request ID
+            Assert.AreEqual(0, update[0].Value<int>()); // Subscription ID
+            Assert.AreEqual(123, update[1].Value<int>()); // Value
```

Comment:
I feel like using something like `FluentAssertions` would make the expectations easier to read.